### PR TITLE
[scheduler cleanup phase 1]: Move NodeInfo utils into pkg/scheduler/cache

### LINKED ISSUE #68958: [scheduler cleanup phase 1]: Move `NodeInfo` utils into pkg/scheduler/cache
**PLEASE COORDINATE WITH SIG-SCHEDULING BEFORE WORKING ON THIS ISSUE.** We want to avoid the situation where two people accidentally start work on the same fix.
Main tracking issue: https://github.com/kubernetes/kubernetes/issues/68951

There are utility functions for use with the `NodeInfo` type in `k8s.io/kubernetes/pkg/scheduler/util`. These should be moved into `k8s.io/kubernetes/pkg/scheduler/cache` because that is where the `NodeInfo` type is defined.

/ref https://github.com/kubernetes/kubernetes/issues/68961
/sig scheduling
/kind cleanup
/help
/good-first-issue

### PR DESCRIPTION
**Pull request by wgliang**
**What this PR does / why we need it**:
Move NodeInfo utils into pkg/scheduler/cache

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #68958

**Special notes for your reviewer**:
/sig scheduler
/ping @misterikkit @k82cn @bsalamat 

**Release note**:

```release-note
Move NodeInfo utils into pkg/scheduler/cache.
```

### CODE REVIEW

File: pkg/scheduler/cache/util.go [hunk 1]
```diff
@@ -76,3 +79,114 @@ func createImageExistenceMap(nodes []*v1.Node) map[string]sets.String {
 	}
 	return imageExistenceMap
 }
+
+// ProtocolPort represents a protocol port pair, e.g. tcp:80.
+type ProtocolPort struct {
```
Comment (misterikkit) [inline pkg/scheduler/cache/util.go]: rather than adding this to utils.go, what do you think about creating a file like ports.go (and ports_test.go)?

Comment (wgliang) [reply pkg/scheduler/cache/util.go]: DONE.

**Review by misterikkit — COMMENTED** [review]
/lgtm

File: pkg/scheduler/cache/ports.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
+/*
```
Comment (bsalamat) [inline pkg/scheduler/cache/ports.go]: Isn't it better to call this file host_ports.go to distinguish it from container ports?

Comment (wgliang) [reply pkg/scheduler/cache/ports.go]: But there also has `ProtocolPort` code, so ports.go maybe more accurate.

Comment (k82cn) [reply pkg/scheduler/cache/ports.go]: +1 to host_ports, we only care about host port in scheduler.

File: pkg/scheduler/cache/util.go [hunk 3]
```diff
@@ -21,6 +21,9 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
+// DefaultBindAllHostIP defines the default ip address used to bind to all host.
```
Comment (k82cn) [inline pkg/scheduler/cache/util.go]: move this const to `ports.go`

**Comment by k82cn** [comment]
LGTM overall

/kind cleanup

File: pkg/scheduler/cache/util.go (hunk 3 above)
Comment (wgliang) [reply pkg/scheduler/cache/util.go]: Ok, I'm fine with that.

**Comment by wgliang** [comment]
/ping @k82cn 
All done.

**Pull request merged**